src/rpc/rawtransaction.cpp

```diff
@@ -1375,178 +1375,27 @@ static void TxCswInErrorToJSON(const CTxCeasedSidechainWithdrawalInput& txcswin,
-    #else
-        const CKeyStore& keystore = tempKeystore;
-    #endif
-
-    int nHashType = SIGHASH_ALL;
-
-    // Script verification errors
-    UniValue vErrors(UniValue::VARR);
-
-    // Sign what we can:
-    for (unsigned int i = 0; i < mergedCert.vin.size(); i++) {
-        CTxIn& txin = mergedCert.vin[i];
-        const CCoins* coins = view.AccessCoins(txin.prevout.hash);
-        if (coins == NULL || !coins->IsAvailable(txin.prevout.n)) {
-            TxInErrorToJSON(txin, vErrors, "Input not found or already spent");
-            continue;
-        }
-        const CScript& prevPubKey = coins->vout[txin.prevout.n].scriptPubKey;
-
-        txin.scriptSig.clear();
-        SignSignature(keystore, prevPubKey, mergedCert, i, nHashType);
-
-        ScriptError serror = SCRIPT_ERR_OK;
-        if (!VerifyScript(
-                txin.scriptSig, prevPubKey, STANDARD_NONCONTEXTUAL_SCRIPT_VERIFY_FLAGS,
-                MutableCertificateSignatureChecker(&mergedCert, i),
-                &serror
-           ))
-        {
-            TxInErrorToJSON(txin, vErrors, ScriptErrorString(serror));
-        }
-    }
-    bool fComplete = vErrors.empty();
-
-    UniValue result(UniValue::VOBJ);
-    result.pushKV("hex", EncodeHexCert(CScCertificate(mergedCert)));
-    result.pushKV("complete", fComplete);
-    if (!vErrors.empty()) {
-        result.pushKV("errors", vErrors);
-    }
-
-    return result;
-}
-
 UniValue signrawtransaction(const UniValue& params, bool fHelp)
 {
     if (fHelp || params.size() < 1 || params.size() > 4)
         throw runtime_error(
             "signrawtransaction \"hexstring\" ( [{\"txid\":\"id\",\"vout\":n,\"scriptPubKey\":\"hex\",\"redeemScript\":\"hex\"},...] [\"privatekey1\",...] sighashtype )\n"
-            "\nSign inputs for raw transaction (serialized, hex-encoded).\n"
+            "\nSign inputs for raw transaction or certificate (serialized, hex-encoded).\n"
+            "Optional arguments for transaction can be follows."
             "The second optional argument (may be null) is an array of previous transaction outputs that\n"
             "this transaction depends on but may not yet be in the block chain.\n"
             "The third optional argument (may be null) is an array of base58-encoded private\n"
             "keys that, if given, will be the only keys used to sign the transaction.\n"
+            "Optional arguments for certificate."
+            "The second optional argument (may be null) is an array of base58-encoded private\n"
+            "keys that, if given, will be the only keys used to sign the transaction.\n"
 #ifdef ENABLE_WALLET
             + HelpRequiringPassphrase() + "\n"
 #endif
 
-            "\nArguments:\n"
-            "1. \"hexstring\"     (string, required) The transaction hex string\n"
-            "2. \"prevtxs\"       (string, optional) An json array of previous dependent transaction outputs\n"
+            "\nArguments for signing transaction:\n"
```

Comment:
Why do we have different interfaces for Txs and Certs?
What is the sense to merge 2 methods in this case?
"prevtxs" is fine to have for certificates.
What about "sighashtype"? need to check, but I think different types can be used for Certificate as well.

Reply:
We discussed with Alberto and decided to unify the interface; the latest changes added the support for `prevtxs` also for the certificates and we enforce that the last parameter (`sighashtype`) is ALL (the other types are not currently supported for certificates).